From e2898a3d1db00dd0e68eb30cd0b6c707731f62e8 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Sun, 2 Jul 2017 23:31:18 +0200
Subject: import: keep volatile keys

We shouldn't wipe volatile keys on container import. Otherwise we'll lose
things like hwaddr and last idmapping.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 lxd/api_internal.go | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/lxd/api_internal.go b/lxd/api_internal.go
index 27157184..3a520be8 100644
--- a/lxd/api_internal.go
+++ b/lxd/api_internal.go
@@ -8,7 +8,6 @@ import (
 	"net/http"
 	"os"
 	"strconv"
-	"strings"
 
 	"github.com/gorilla/mux"
 	"gopkg.in/yaml.v2"
@@ -345,11 +344,6 @@ func internalImport(d *Daemon, r *http.Request) Response {
 	}
 
 	baseImage := backup.Container.Config["volatile.base_image"]
-	for k := range backup.Container.Config {
-		if strings.HasPrefix(k, "volatile") {
-			delete(backup.Container.Config, k)
-		}
-	}
 
 	arch, err := osarch.ArchitectureId(backup.Container.Architecture)
 	if err != nil {
@@ -430,11 +424,6 @@ func internalImport(d *Daemon, r *http.Request) Response {
 		}
 
 		baseImage := snap.Config["volatile.base_image"]
-		for k := range snap.Config {
-			if strings.HasPrefix(k, "volatile") {
-				delete(snap.Config, k)
-			}
-		}
 
 		arch, err := osarch.ArchitectureId(snap.Architecture)
 		if err != nil {
